### ISSUE TITLE
ser2net support

### ISSUE THREAD

**Issue by rossiniscarface** [ioBroker/ioBroker.rflink#13]
Hi,

es wäre toll wenn man den rflink adapter auch über tcp mit ser2net konfigurieren könnte. 
mfg

**Comment by Apollon77** [comment]
Hm... why you do not do net2ser and use this ?!

**Comment by rossiniscarface** [comment]
Could you point me to the documentation for net2ser?
I could not find it anywhere.

**Comment by Apollon77** [comment]
Ok, ich gehe davon aus das Du mit "ser2net" meinst das der Serielle Port auf einem entfernten Rechner läuft und d kein ioBroker-Slave möglich ist. Korrekt?
Und du würdest gern auf dem Rechner mit "ser2net" den seriellen Port per Netzwerk zur verfügung stellen ... naja dann brauchst Du auf dem ioBroker.Rechner das "gegenstück" um wieder einen lokalen seriellen port zu simulieren. Ist das bei der Software nicht dabei?

Ich weiss das generell "socat" sowas kann.

**Comment by Apollon77** [comment]
PS: Das einfachste ist aber einfach da wo der serielle Port ist einen ioBroker Slave zu machen und dort den Adapter laufen zu lassen :-)

**Comment by rossiniscarface** [comment]
Hi,

also ser2net ist ein kleiner Linux-Dienst, der einen Seriellen Adapter über TCP bereitstellt. Den kann man dann am Client-System ganz einfach mit Telnet konsumieren und das funktioniert auch.
Siehe [man(8) ser2net](https://linux.die.net/man/8/ser2net)
Im FAQ von RFLINK steht dass man ser2net für Netzwerk verwenden soll. [siehe](http://www.rflink.nl/blog2/faq#Network)

Ich habe jetzt nach einer ganzen Nacht einiges an Software gefunden die jetzt aus dem Netzwerkstream wieder einen virtuellen Seriellen port macht der dann im 
Also mit socat pty,link=/dev/ttyS3 tcp:REMOTEIP:5000
oder mit cyclates-serial-client mit folgender config : 
/dev/ttyS3:prts:REMOTEIP:1:rtelnet: -m 1 -p 4999

Damit klappt es jetzt mit dem RFLink.

PS.: einen kompletten ioBroker Slave gibts wahrscheinlich nicht für OpenWRT oder? Da hab ich den rfLink mit ser2net laufen.
PPS.: das rflink gateway in Domoticz kann einfach mit IP und Port auf ser2net zugreifen.

Warum steck ich eigentlich den RFLink nicht direkt an den ioBroker? Weil der in einem HyperV VM läuft :)

**Comment by Apollon77** [comment]
Ich experimentiere mit meinen vms gerade mit usbip ... ist in vielen linuxen schon standardmäßig dabei.

**Comment by rossiniscarface** [comment]
Ja, hab ich mich vor Jahren schon mal damit gespielt, aber das war mir zu instabil, bzw. hat die Potenz den Kernel in Mitleidenschaft zu ziehen.

**Comment by Jennobert** [comment]
Hallo.
Sorry wenn ich das alte issue hier noch mal aufwärme. Aber ich suche mittlerweile auch schon seid Monaten immer mal wieder nach einer Lösung um das RFLink Interface an meinen entfernten (im Keller auf Proxmox) ioBroker über den RFLink-Adapter zu koppeln. 

Mit socrat komme ich nur soweit das die Schnittstelle wohl vorhanden ist, aber im ioBroker im RFLink Adapter nicht angezeigt wird.

@rossiniscarface kannst du mir da vielleicht ein wenig auf die Sprünge helfen ?

**Comment by rossiniscarface** [comment]
Sorry aber ich hab die Lösung wieder abgebaut. 
Mit Ser2Net war das alles instabil. Hab dann auf einem zweiten Rechner eine Slave-Instanz von iobroker am laufen und da war auch das Problem, dass iobroker nicht zuverlässig die rflink signale verarbeitet hat.

**Comment by Jennobert** [comment]
Danke für die Antwort. 
Dann geht´s eben doch wieder auf die Variante mit MQTT zurück.

https://github.com/seb821/espRFLinkMQTT
